examples/pytorch/grand/utils.py

```diff
@@ -0,0 +1,17 @@
+import torch as th
+import torch.nn as nn
+import torch.nn.functional as F
+
+def consis_loss(logps, temp, lam):
+    ps = [th.exp(p) for p in logps]
+    sum_p = 0.
+    for p in ps:
```

Comment:
It seems this part does not fully leverage pytorch's vectorized operation. Please consider tune for loops into some pytorch tensor operations.